### DISCUSSION TITLE
Remix link router not routing

### DISCUSSION THREAD

**Discussion by xxCosta** [remix-run/remix#7192] [Q&A]
I have a simple 3 page application. When i click the <Link> 1 of my 3 pages doesn't load. The route that does not link has a loader on it while the other 2 don't. When i click the link, i get a refreshed page of the current page im on. However, if I manually type in the url, it loads the page. When i click the link i get this response in the console `GET /beats?_data=routes%2Fbeats._index 200` where i should get just `GET /beats`. More info and video of what's happening can be found on this discord thread: https://discord.com/channels/770287896669978684/1140811143087132702

**Comment by kiliman** [comment]
I tested it and was able to get it to work once I commented out the line `import "dotenv/config"`. This should be moved to `entry.server.jsx`. Also when returning data from loaders, you should use the `json` helper from `"@remix-run/node"`

```ts
//import "dotenv/config";
import { createClient } from "redis";
import { useLoaderData } from "@remix-run/react";
//import Sketch from 'react-p5';
//import 'p5/lib/addons/p5.sound';
import { Schema, Repository, Client, EntityId } from "redis-om";
import { json } from "@remix-run/node";

export async function loader() {
  // const connection = createClient({
  //     url: `redis://${process.env.REDIS_ADMIN_USERNAME}:${process.env.REDIS_ADMIN_PASS}@127.0.0.1:6379`
  // });
  // await connection.connect()
  // const client = await new Client().use(connection)

  // const beatSchema = new Schema('beat', {
  //   name: {type: 'string'},
  //   producer: {type: 'string'},
  //   genre: {type: 'string'},
  //   bpm: {type: 'number'},
  //   tags: {type: 'string[]'},
  //   plays: {type: 'number'},
  //   id: {type: 'number'}
  // },{dataStructure: 'HASH'})

  // const beatRepository = new Repository(beatSchema, client)
  // await beatRepository.createIndex()
  // const beats = await beatRepository.search().return.all()
  // //const beat = await beatRepository.fetch(beatId)
  // let retBeats = beats
  // await connection.disconnect()
  // console.log(connection.isOpen)
  // return (retBeats)
  const beats = [
    { id: 1, name: "test1" },
    { id: 2, name: "test2" },
    { id: 3, name: "test3" },
  ];
  console.log("beats loader", beats);
  return json(beats);
}
```
<img width="589" alt="image" src="https://github.com/remix-run/remix/assets/47168/01e6dd9e-0d4a-4972-ae00-3f25fc7e0bd7">

<img width="796" alt="image" src="https://github.com/remix-run/remix/assets/47168/e9cb8dd3-c3a0-450b-982c-654520942811">

**Reply by xxCosta** [reply]
https://github.com/xxCosta/69beats/commit/fde858c70595f7e21ab92f19f7e8d38f31296fe2 issue is still present after making these changes. "redis-om" already returns a json object so no need to call json again in the return. I think it's a problem with redis or the "redis-node/redis-on" library as remix by defualt uses node 14 which doesnt include features such as top level await. Currently trying to figure out how to change the default node version in remix, some others have had trouble with this issue as well: https://github.com/remix-run/remix/issues/4348

**Reply by xxCosta** [reply]
https://github.com/xxCosta/69beats/commit/aeadc769829900f603836b3fcf9a37089edc6a4e Issue has someting to do with redis library. I returned the response I get from redis as a variable and it works flawlessly. Will start issue in redis discord.

**Reply by kiliman** [reply]
The minimum version supported Node version is v14, but I believe the recommended version is v18+.

When you say "_json object_" do you mean an actual JavaScript object, or a string from `JSON.stringify`? Although Remix will automatically convert a JS object to a JSON string for you, it is considered best practice to always use a helper function, like `json`, `defer`, etc. As these return actual `Response` objects, not just a naked object.

**Reply by xxCosta** [reply]
you're correct it is a javascrip object, i'll use `json()` from now on. Issue is with redis library, i'll change databases and try again.
